services/121-service/src/registration/services/registrations-bulk.service.spec.ts

```diff
@@ -34,18 +34,23 @@ describe('RegistrationBulkService', () => {
     jest
       .spyOn(messageTemplateRepository as any, 'findOne')
       .mockImplementation((arg: any) => {
-        if (arg.where.type === RegistrationStatusEnum.registered) {
+        const programIdValue = arg.where.programId._value;
+        const typeValue = arg.where.type._value;
+
+        if (
+          programIdValue === programId &&
+          typeValue === RegistrationStatusEnum.registered
+        ) {
           return Promise.resolve({
             id: 1,
             name: 'test',
             language: LanguageEnum.en,
-            type: 'test',
-            message: 'test',
+            type: RegistrationStatusEnum.registered,
```

Comment:
Why this change? It seems like this is less explicit and more prone to failures but I'm guessing there is a reason I cannot see. I'm also surprised this doesn't break any existing tests 🤔